libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -38,17 +38,52 @@ namespace
 {
 
 template <class T, class B>
-bool hasEqualNameAndParameters(T const& _a, B const& _b)
+bool hasEqualParameters(T const& _a, B const& _b)
 {
-	return
-		_a.name() == _b.name() &&
-		FunctionType(_a).asExternallyCallableFunction(false)->hasEqualParameterTypes(
-			*FunctionType(_b).asExternallyCallableFunction(false)
-		);
+	return FunctionType(_a).asExternallyCallableFunction(false)->hasEqualParameterTypes(
+		*FunctionType(_b).asExternallyCallableFunction(false)
+	);
 }
 
 }
 
+template<typename T>
+map<ASTString, vector<T>> ContractLevelChecker::filterDeclarationsInSourceUnit(
```

Comment:
```suggestion
map<ASTString, vector<T>> ContractLevelChecker::filterDeclarations(
```

Reply:
Actually does it need to be a member function or can it just be a free function in an anonymous namespace?